File: Python/21-30/e_29.py
# Escribe un programa en python que permita recuperar la extensión de un archivo.


import os

def obtener_extension(nombre_archivo):
    # Dividir el nombre del archivo y la extensión
    nombre_base, extension = os.path.splitext(nombre_archivo)
    
    # Eliminar el punto de la extensión
    extension = extension[1:]
    
    return extension

# Ejemplo de uso
nombre_archivo = "mi_archivo.txt"
extension = obtener_extension(nombre_archivo)

print(f"La extensión del archivo '{nombre_archivo}' es: {extension}")


# Puedes utilizar la biblioteca os.path en Python para obtener la extensión de un archivo.
# Este programa define una función llamada obtener_extension que toma el nombre de un archivo,
# utiliza os.path.splitext para dividir el nombre del archivo y la extensión, y luego elimina el punto de la extensión. 
# Finalmente, se imprime la extensión del archivo. Puedes cambiar el valor de nombre_archivo por el nombre del archivo que desees analizar.

File: JS/21-30/e_29.js
function obtenerExtension(nombreArchivo) {
    // Obtener la última aparición del punto para separar la extensión
    const puntoIndex = nombreArchivo.lastIndexOf('.');
    
    // Verificar si hay un punto en el nombre del archivo
    if (puntoIndex !== -1) {
        // Obtener la subcadena después del último punto
        const extension = nombreArchivo.slice(puntoIndex + 1);
        return extension;
    } else {
        // Si no hay un punto, el archivo no tiene extensión
        return "El archivo no tiene extensión";
    }
}

// Ejemplo de uso
const nombreArchivo = "mi_archivo.txt";
const extension = obtenerExtension(nombreArchivo);

console.log(`La extensión del archivo '${nombreArchivo}' es: ${extension}`);


// Este programa define una función llamada obtenerExtension que toma el nombre de un archivo,
// encuentra la última aparición del punto (que separa la extensión del nombre del archivo) y luego obtiene la subcadena que representa la extensión.
// Si no se encuentra un punto en el nombre del archivo, se imprime un mensaje indicando que el archivo no tiene extensión.
// Puedes cambiar el valor de nombreArchivo por el nombre del archivo que desees analizar.